src/shared/components/drop-target.jsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "react-emotion";
+
+const StyledDropTarget = styled.div`
+  align-content: center;
+  align-items: center;
+  background: #ededed;
+  border-radius: 5px;
+  box-sizing: border-box;
+  display: grid;
+  justify-content: center;
+  justify-items: center;
+  padding: 20px;
+
+  width: ${props => (props.width ? `${props.width}px` : "100%")};
+
+  /* Specify an exact height so that the target doesn't grow or shrink when the
+     hover message is displayed */
+  height: ${props => (props.height ? `${props.height}px` : "100px")};
+
+  &.hovered {
+    border: 3px dashed #333;
+  }
+`;
+
+export default class DropTarget extends React.Component {
+  static propTypes = {
+    children: PropTypes.oneOfType([
+      PropTypes.arrayOf(PropTypes.node),
+      PropTypes.node
+    ]),
+    getHoverMessage: PropTypes.func,
```

Comment:
`getHoverMessage` feels narrowly useful (just the number of files). Seems like it could just be a function that returns the array of files, and the user can return whatever they want (the total number, the names, return a bunch of React components that are to be rendered here). Could you do a pass where it is a bit more flexible?

With that in mind, it does seem like we could here return either a function (which for example could return a React component if we need access to the file contents / metadata) or a rendered React component.